--- a/chrome/browser/ui/tabs/tab_style.cc
+++ b/chrome/browser/ui/tabs/tab_style.cc
@@ -17,17 +17,17 @@ namespace {
 
 // Thickness in DIPs of the separator painted on the left and right edges of
 // the tab.
-constexpr int kChromeRefreshSeparatorThickness = 1;
-constexpr int kChromeRefreshSeparatorHorizontalMargin = 5;
+constexpr int kChromeRefreshSeparatorThickness = 2;
+constexpr int kChromeRefreshSeparatorHorizontalMargin = 2;
 // TODO (crbug.com/1451400): This constant should be in LayoutConstants.
 constexpr int kChromeRefreshSeparatorHeight = 16;
 
 // The padding from the top of the tab to the content area.
 constexpr int kChromeRefreshTabVerticalPadding = 6;
-constexpr int kChromeRefreshTabHorizontalPadding = 2;
+constexpr int kChromeRefreshTabHorizontalPadding = 8;
 
 // The standard tab width is 232 DIP, excluding separators and overlap.
-constexpr int kTabWidth = 190;
+constexpr int kTabWidth = 232;
 
 // Split tabs need more space to be readable
 constexpr int kSplitTabWidth = 250;
@@ -41,7 +41,7 @@ int TabStyle::GetStandardWidth(const boo
   if (is_split) {
     // Split tabs appear as half width with one bottom extension. They also must
     // include half the tab overlap as the tabs fill the space between them.
-    return kSplitTabWidth / 2 + GetBottomCornerRadius() + GetTabOverlap() / 2;
+    return kTabWidth / 2 + GetBottomCornerRadius() + GetTabOverlap() / 2;
   } else {
     // The full width includes two extensions with the bottom corner radius.
     return kTabWidth + 2 * GetBottomCornerRadius();
@@ -92,11 +92,11 @@ int TabStyle::GetMinimumInactiveWidth()
 }
 
 int TabStyle::GetTopCornerRadius() const {
-  return 8;
+  return 10;
 }
 
 int TabStyle::GetBottomCornerRadius() const {
-  return 8;
+  return 12;
 }
 
 int TabStyle::GetTabOverlap() const {
@@ -135,12 +135,9 @@ std::tuple<float, float, float, SkColor>
   const SkColor inactive_bg =
       GetTabBackgroundColor(TabStyle::TabSelectionState::kInactive,
                             /*hovered=*/false, frame_active, color_provider);
-  const SkColor inactive_contrast_bg = color_utils::GetResultingPaintColor(
-      inactive_bg, color_provider->GetColor(kColorToolbar));
-  const auto get_blend = [inactive_contrast_bg](SkColor target,
-                                                float contrast) {
-    return color_utils::BlendForMinContrast(
-        inactive_contrast_bg, inactive_contrast_bg, target, contrast);
+  const auto get_blend = [inactive_bg](SkColor target, float contrast) {
+    return color_utils::BlendForMinContrast(inactive_bg, inactive_bg, target,
+                                            contrast);
   };
 
   const SkColor active_bg =
@@ -271,9 +268,11 @@ SkColor TabStyle::GetCurrentTabBackgroun
 
 gfx::Insets TabStyle::GetContentsInsets() const {
   return gfx::Insets::TLBR(
-      kChromeRefreshTabVerticalPadding,
+      kChromeRefreshTabVerticalPadding +
+          GetLayoutConstant(LayoutConstant::kTabStripPadding),
       GetBottomCornerRadius() + kChromeRefreshTabHorizontalPadding,
-      kChromeRefreshTabVerticalPadding,
+      kChromeRefreshTabVerticalPadding +
+          GetLayoutConstant(LayoutConstant::kTabStripPadding),
       GetBottomCornerRadius() + kChromeRefreshTabHorizontalPadding);
 }
 
@@ -202,8 +202,6 @@ SkPath TabStyleViewsImpl::GetPath(TabSty
   CHECK(tab());
   const int stroke_thickness = GetStrokeThickness(flags.force_active);
 
-  const TabStyle::TabSelectionState state = GetSelectionState();
-
   // We'll do the entire path calculation in aligned pixels.
   // TODO(dfried): determine if we actually want to use `stroke_thickness` as
   // the inset in this case.
@@ -214,14 +212,18 @@ SkPath TabStyleViewsImpl::GetPath(TabSty
   // tab width (in DIP), not our new, scaled-and-aligned bounds.
   float content_corner_radius =
       GetTopCornerRadiusForWidth(tab()->width()) * scale;
-  float extension_corner_radius = 4 * scale;
+  float extension_corner_radius = tab_style()->GetBottomCornerRadius() * scale;
 
   const float separator_overlap = (tab_style()->GetSeparatorMargins().width() +
                                    tab_style()->GetSeparatorSize().width()) *
                                   scale;
 
   // Selected, hover, and inactive tab fills are a detached squarcle tab.
-  if ((true)) {
+  if ((path_type == TabStyle::PathType::kFill &&
+       state != TabStyle::TabSelectionState::kActive) ||
+      path_type == TabStyle::PathType::kHighlight ||
+      path_type == TabStyle::PathType::kInteriorClip ||
+      path_type == TabStyle::PathType::kHitTest) {
     float top_left_corner_radius = content_corner_radius;
     float top_right_corner_radius = content_corner_radius;
     float bottom_left_corner_radius = content_corner_radius;
@@ -268,6 +270,39 @@ SkPath TabStyleViewsImpl::GetPath(TabSty
       top_right_corner_radius = 0;
     }
 
+    // While the tab is closing do not add extra space as it degrades the close
+    // tab annimation.
+    if (!tab()->closing()) {
+      // If the size of the space for the path is smaller than the size of a
+      // favicon, if we are building a path for the hit test, or if we are
+      // building a path for a split tab, expand to take the entire width of the
+      // separator margins AND the separator.
+      const bool limited_tab_space =
+          (right - left) < (gfx::kFaviconSize * scale);
+      const bool expand_into_left_separator =
+          limited_tab_space || path_type == TabStyle::PathType::kHitTest ||
+          IsRightSplitTab(tab());
+      const bool expand_into_right_separator =
+          limited_tab_space || path_type == TabStyle::PathType::kHitTest ||
+          IsLeftSplitTab(tab());
+      if (expand_into_left_separator) {
+        left -= separator_overlap / 2.0;
+      }
+      if (expand_into_right_separator) {
+        right += separator_overlap / 2.0;
+      }
+    }
+
+    if (tab()->split().has_value()) {
+      if (IsLeftSplitTab(tab())) {
+        top_right_corner_radius = 0;
+        bottom_right_corner_radius = 0;
+      } else if (IsRightSplitTab(tab())) {
+        top_left_corner_radius = 0;
+        bottom_left_corner_radius = 0;
+      }
+    }
+
     // Radii are clockwise from top left.
     const SkVector radii[4] = {
         SkVector(top_left_corner_radius, top_left_corner_radius),
@@ -510,6 +545,19 @@ gfx::Insets TabStyleViewsImpl::GetConten
   gfx::Insets base_style_insets = tab_style()->GetContentsInsets();
   gfx::Insets split_insets = gfx::Insets(0);
 
+  // For split tabs, remove insets equal to the total separator width between
+  // them.
+  const float total_separator_width =
+      tab_style()->GetSeparatorMargins().left() +
+      tab_style()->GetSeparatorSize().width() +
+      tab_style()->GetSeparatorMargins().right();
+  if (IsRightSplitTab(tab())) {
+    split_insets.set_left(total_separator_width / -2);
+  }
+  if (IsLeftSplitTab(tab())) {
+    split_insets.set_right(total_separator_width / -2);
+  }
+
   return gfx::Insets::TLBR(
              0, 0, GetLayoutConstant(LayoutConstant::kTabstripToolbarOverlap),
              0) +
@@ -792,6 +840,15 @@ float TabStyleViewsImpl::GetHoverOpacity
 }
 
 int TabStyleViewsImpl::GetStrokeThickness(bool should_paint_as_active) const {
+  std::optional<tab_groups::TabGroupId> group = tab_->group();
+  if (group.has_value() && tab_->IsActive()) {
+    return TabGroupUnderline::kStrokeThickness;
+  }
+
+  if (tab_->IsActive() || should_paint_as_active) {
+    return tab_->controller()->GetStrokeThickness();
+  }
+
   return 0;
 }
 
@@ -821,7 +878,9 @@ SkColor TabStyleViewsImpl::GetTabSeparat
     return gfx::kPlaceholderColor;
   }
 
-  return cp->GetColor(kColorToolbarExtensionSeparatorEnabled);
+  return cp->GetColor(tab()->GetWidget()->ShouldPaintAsActive()
+                          ? kColorTabDividerFrameActive
+                          : kColorTabDividerFrameInactive);
 }
 
 SkColor TabStyleViewsImpl::GetCurrentTabBackgroundColor(
@@ -870,7 +929,8 @@ void TabStyleViewsImpl::PaintTabBackgrou
   const auto* widget = tab_->GetWidget();
   DCHECK(widget);
   const SkColor tab_stroke_color = widget->GetColorProvider()->GetColor(
-      kColorToolbarActionItemEngaged);
+      tab_->GetWidget()->ShouldPaintAsActive() ? kColorTabStrokeFrameActive
+                                               : kColorTabStrokeFrameInactive);
 
   PaintBackgroundStroke(canvas, selection_state,
                         group_color.value_or(tab_stroke_color));
